tests/io_components/test_read.py

```diff
@@ -57,10 +57,10 @@ def test_dispatch_metapartitions_without_dataset_metadata(dataset, store_session
     assert len(partitions) == 2
 
     mp = partitions[0]
-    assert mp.dataset_metadata == {}
+    assert list(mp.dataset_metadata.keys()) == ["creation_time"]
```

Comment:
weird, this is a regression. We changed the dispatch to no longer include the metadata with the MPs since this redundant data duplication was causing issues. I can't understand which change affected this behaviour.